static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -53,6 +78,193 @@ class SpanTreeModel {
       )
     );
   }
+
+  isSpanFilteredOut = (props: {
+    operationNameFilters: ActiveOperationFilter;
+    filterSpans: FilterSpans | undefined;
+  }): boolean => {
+    const {operationNameFilters, filterSpans} = props;
+
+    if (operationNameFilters.type === 'active_filter') {
+      const operationName = getSpanOperation(this.span);
+
+      if (
+        typeof operationName === 'string' &&
+        !operationNameFilters.operationNames.has(operationName)
+      ) {
+        return true;
+      }
+    }
+
+    if (!filterSpans) {
+      return false;
+    }
+
+    return !filterSpans.spanIDs.has(getSpanID(this.span));
+  };
+
+  generateSpanGap(
+    event: Readonly<EventTransaction>,
+    previousSiblingEndTimestamp: number | undefined,
+    treeDepth: number,
+    continuingTreeDepths: Array<TreeDepthType>
+  ): EnhancedProcessedSpanType | undefined {
+    // hide gap spans (i.e. "missing instrumentation" spans) for browser js transactions,
+    // since they're not useful to indicate
+    const shouldIncludeGap = !isEventFromBrowserJavaScriptSDK(event);
+
+    const isValidGap =
+      typeof previousSiblingEndTimestamp === 'number' &&
+      previousSiblingEndTimestamp < this.span.start_timestamp &&
+      // gap is at least 100 ms
+      this.span.start_timestamp - previousSiblingEndTimestamp >= 0.1 &&
+      shouldIncludeGap;
```

Comment:
Nit: Maybe move `shouldIncludeGap` to the beginning of the expression since none of the other conditions matter if `shouldIncludeGap` is false

Reply:
Done in https://github.com/getsentry/sentry/pull/26746/commits/7e56bde8ddc2e84394074383be63f290826104a8